### DISCUSSION TITLE
Using Prisma Logger in Nest.js

### DISCUSSION THREAD

**Discussion by bnhr** [prisma/prisma#14743] [Q&A] [answered]
Folks, is there any working example of how to create query and error logging for Prisma in Nest.js project?

**Comment by ludralph** [comment]
Hi @bnhr 👋 

To log queries using Prisma,  do like below

```js
 const prisma = new PrismaClient({
  log: ['query', 'info', 'warn', 'error'],
})
})
```

To learn more about logging in Prisma, check out the [docs on logging](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging)

**Reply by bnhr** [reply]
I've read it, and implemented like this:

```
export class PrismaService extends PrismaClient implements OnModuleInit {
	constructor() {
		super({
			log: [
				{
					emit: 'event',
					level: 'query',
				},
				{
					emit: 'event',
					level: 'error',
				},
				{
					emit: 'stdout',
					level: 'info',
				},
				{
					emit: 'stdout',
					level: 'warn',
				},
			],
		})
	}

	async onModuleInit() {
		await this.$connect()
	}

	async enableShutdownHooks(app: INestApplication) {
		this.$on('beforeExit', async (e) => {
			console.log(e.name)
			await app.close()
		})
	}
}
```

But the log is not printed..

**Reply by maximkoev** [reply]
you emit event but do not listen event

**Comment by bnhr** [comment, marked as answer]
I've found the answer:

```
import { INestApplication, Injectable, OnModuleInit } from '@nestjs/common'
import { PrismaClient } from '@prisma/client'

@Injectable()
export class PrismaService extends PrismaClient implements OnModuleInit {
	constructor() {
		super({
			log: ['query', 'info', 'warn', 'error'],
			errorFormat: 'pretty',
		})
	}

	async onModuleInit() {
		await this.$connect()
	}

	async enableShutdownHooks(app: INestApplication) {
		this.$on('beforeExit', async (e) => {
			console.log(e.name)
			await app.close()
		})
	}
}
```

**Reply by sfedorenko** [reply]
Try this:
```
@Module({
	imports: [
		PrismaModule.forRoot({
			isGlobal: true,
			prismaServiceOptions: {
				prismaOptions: {
					log: [
						{ emit: 'stdout', level: 'query' },
					]
				}
			}
		}),
	],
	...
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
